contracts/UsingHelpers.sol

```diff
@@ -2,111 +2,100 @@
 
 pragma solidity 0.7.6;
 
-/**
- * @title Utility contract to enable access to common structures
- */
-contract UsingHelpers {
-    uint8 internal constant ONE = 1;
-    uint8 internal constant IDX_COMMIT = 7;
-    uint8 internal constant IDX_REDEEM = 6;
-    uint8 internal constant IDX_REFUND = 5;
-    uint8 internal constant IDX_EXPIRE = 4;
-    uint8 internal constant IDX_COMPLAIN = 3;
-    uint8 internal constant IDX_CANCEL_FAULT = 2;
-    uint8 internal constant IDX_FINAL = 1;
+// Those are the payment methods we are using throughout the system.
+// Depending on how to user choose to interact with it's funds we store the method, so we could distribute its tokens afterwise
+enum PaymentMethod {
+    ETHETH,
+    ETHTKN,
+    TKNETH,
+    TKNTKN
+}
 
-    /*  Status of the voucher in 8 bits:
-        [7:COMMITTED] [6:REDEEMED] [5:REFUNDED] [4:EXPIRED] [3:COMPLAINED] [2:CANCELORFAULT] [1:FINAL] [1:/]
-    */
+enum VoucherState {FINAL, CANCEL_FAULT, COMPLAIN, EXPIRE, REFUND, REDEEM, COMMIT}
+/*  Status of the voucher in 8 bits:
+    [6:COMMITTED] [5:REDEEMED] [4:REFUNDED] [3:EXPIRED] [2:COMPLAINED] [1:CANCELORFAULT] [0:FINAL]
+*/
 
-    // Those are the payment methods we are using throughout the system.
-    // Depending on how to user choose to interact with it's funds we store the method, so we could distribute its tokens afterwise
-    uint8 internal constant ETHETH = 1;
-    uint8 internal constant ETHTKN = 2;
-    uint8 internal constant TKNETH = 3;
-    uint8 internal constant TKNTKN = 4;
+uint8 constant ONE = 1;
 
-    struct VoucherDetails {
-        uint256 tokenIdSupply;
-        uint256 tokenIdVoucher;
-        address issuer;
-        address holder;
-        uint256 price;
-        uint256 depositSe;
-        uint256 depositBu;
-        uint256 price2pool;
-        uint256 deposit2pool;
-        uint256 price2issuer;
-        uint256 deposit2issuer;
-        uint256 price2holder;
-        uint256 deposit2holder;
-        uint8 paymentMethod;
-        VoucherStatus currStatus;
-    }
+struct VoucherDetails {
+    uint256 tokenIdSupply;
+    uint256 tokenIdVoucher;
+    address issuer;
+    address holder;
+    uint256 price;
+    uint256 depositSe;
+    uint256 depositBu;
+    uint256 price2pool;
+    uint256 deposit2pool;
+    uint256 price2issuer;
+    uint256 deposit2issuer;
+    uint256 price2holder;
+    uint256 deposit2holder;
+    PaymentMethod paymentMethod;
+    VoucherStatus currStatus;
+}
 
-    struct VoucherStatus {
-        uint8 status;
-        bool isPaymentReleased;
-        bool isDepositsReleased;
-        uint256 complainPeriodStart;
-        uint256 cancelFaultPeriodStart;
-    }
+struct VoucherStatus {
+    uint8 status;
```

Comment:
Shouldn't this also be of type VoucherStatus?

Reply:
No, `VoucherState` has values [0,6], and represent exactly one state.
While `status` is combined status of all states that voucher has been through (subset of values on [0 and 64])

Reply:
OK